aws/aws.go

```diff
@@ -672,6 +702,20 @@ func getImages(ec2Clients map[string]ec2Client) ([]*types.Image, error) {
 	return images, nil
 }
 
+func getDisk(ec2Client ec2Client, volumeId string) (*types.Disk, error) {
+
+	result, err := ec2Client.DescribeVolumes(&ec2.DescribeVolumesInput{VolumeIds: []*string{&volumeId}})
+	if err != nil {
+		log.Errorf("[AWS] Failed to fetch the volume, err: %s", err)
+		return nil, err
+	}
+	log.Debugf("[AWS] Processing volumes (%d): [%s]", len(result.Volumes), result.Volumes)
+	for _, vol := range result.Volumes {
+		return newDisk(vol), nil
```

Comment:
This loop processes only the first element. Why not use `result.Volumes[0]` instead?

Reply:
I did not wanted to cause an "index out of range" error - there is a more readable one at the end of the method.

Reply:
While `*types.Disk` should be nil, this can solve it:
```
var resp *types.Disk
if len(result.Volumes) > 0 {
    resp = result.Volumes[0]
}
return resp, nil
```

Reply:
done